framework/Collections/TList.php

```diff
@@ -76,15 +78,30 @@ public function __construct($data = null, $readOnly = false)
 	 */
 	public function getReadOnly(): bool
 	{
-		return $this->_r;
+		return (bool) $this->_r;
 	}
 
 	/**
-	 * @param bool $value whether this list is read-only or not
+	 * @param null|bool|string $value whether this list is read-only or not
 	 */
-	protected function setReadOnly($value)
+	public function setReadOnly($value)
 	{
-		$this->_r = TPropertyValue::ensureBoolean($value);
+		if ($value === null) {
+			return;
+		}
+		if($this->_r === null || Prado::isCallingSelf()) {
+			$this->_r = TPropertyValue::ensureBoolean($value);
+		} else {
+			throw new TInvalidOperationException('list_readonly_set', $this::class);
+		}
+	}
+
+	/**
+	 * This sets the read only property.
+	 */
+	protected function ensureReadOnly(): void
+	{
+		$this->_r = (bool) $this->_r;
```

Comment:
Do i get this correctly, is this this used to force null to false if unset, or leave the actual value if already set?
The name ensureReadOnly() maybe is  a bit misleading, i would assume `$this->_r = true` given the name

Reply:
yes.  it's to "collapse" the ReadOnly from the null state.   I really didn't know what to call the function.  Can you suggest a better function name?  I'm at a bit of a loss.

Reply:
Sometimes, we get so lost in the function and functionality that the name becomes an oversight.  Kinda like the new "method_visible" aka new "method_exists".

Reply:
ok.  I'm thinking "collapseReadOnly".

Reply:
also, sorry about changing the name of method_accessible again.  I do think it is better as method_visible as that is what it computes....  PHP Visibility.

Reply:
Quite had to name it, to be honest, since it will also "pollute" the child classes. ensureReadOnlyIsInitialized(), ensureReadOnlyHasValue() seems too complex.. a different verb like reduceReadOnly() could be used, even if reduce is usually associated to arrays.. i'm not sure.

Reply:
collapseReadOnly sounds great! +1 also for method_visible